README.md

```diff
@@ -98,7 +98,10 @@ The following lists all the class parameters the `bacula` class accepts.
 
 For directors, `clients` is a hash of clients.  The keys are the clients while
 the values are a hash of parameters. The parameters accepted are the same as
-the `bacula::client::config` define.
+the `bacula::client::config` define. But allows additionally to enable encryption
+through setting `pki_encryption` to true and defining `pki_keypair` and
+`pki_master_key` files (which have to be placed on the client first).
```

Comment:
This would be a good opportunity to collect an exported resource from the bacula director.  For example, on the master node:

@@file { '/var/lib/bacula/ssl/certs/master.crt':
    tag        => 'bacula_master_key',
    content => file("/var/lib/bacula/ssl/certs/${fqdn}.pem"),
}

This resource would then be *realized* on the bacula client nodes as folllows.

```
File <<| tag == 'bacula_master_key' |>>
```

Reply:
The master key isn't necessarily the cert from the director. I created a completely new keypair for that and am delivering just the public key to all "clients". This is only for disaster recovery if the key on the client isn't available any more so the backup can still be decrypted. I wouldn't use the directors key for that.

Reply:
Okay, that makes sense.  My other concern would be how is the master key file managed?  Is it created outside of puppet?

Reply:
Yes, completely. The point is that the private key for that keypair should be kept as private as possible as it allows to decrypt any backup.
That's the reason why I created the client keypair directly on every client. So the keypair is only available on that specific client.

I created the master keypair on my local machine. The public key is delivered to every client trough puppet (stored in hiera) using a custom ssl-cert module/profile.